doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst

```diff
@@ -0,0 +1,61 @@
+
+   >>> from sympy import symbols, Matrix, solve, simplify
+   >>> from sympy.physics.mechanics import *
+   >>> q1, q2, q3, u1, u2, u3 = dynamicsymbols('q1:4, u1:4')
+   >>> l1, l2, l3, l4, rho = symbols('l1:5, rho')
+
+With all symbols defined, we can now define the bodies. ::
+
+   >>> N = ReferenceFrame('N')
+   >>> inertias = [inertia(N, 0, 0, rho * l ** 3 / 12) for l in (l1, l2, l3, l4)]
+   >>> link1 = Body('Link1', frame=N, mass=rho * l1, central_inertia=inertias[0])
+   >>> link2 = Body('Link2', mass=rho * l2, central_inertia=inertias[1])
+   >>> link3 = Body('Link3', mass=rho * l3, central_inertia=inertias[2])
+   >>> link4 = Body('Link4', mass=rho * l4, central_inertia=inertias[3])
+
+Next, we also define the first three joints. ::
+
+   >>> joint1 = PinJoint('J1', link1, link2, coordinates=q1, speeds=u1,
+   ...                   parent_axis=link1.z, parent_joint_pos=l1 / 2 * link1.x,
+   ...                   child_axis=link2.z, child_joint_pos=-l2 / 2 * link2.x)
+   >>> joint2 = PinJoint('J2', link2, link3, coordinates=q2, speeds=u2,
+   ...                   parent_axis=link2.z, parent_joint_pos=l2 / 2 * link2.x,
+   ...                   child_axis=link3.z, child_joint_pos=-l3 / 2 * link3.x)
+   >>> joint3 = PinJoint('J3', link3, link4, coordinates=q3, speeds=u3,
+   ...                   parent_axis=link3.z, parent_joint_pos=l3 / 2 * link3.x,
+   ...                   child_axis=link4.z, child_joint_pos=-l4 / 2 * link4.x)
+
+Now we can formulate the holonomic constraint that will close the kinematic loop. ::
+
+   >>> loop = link4.masscenter.pos_from(link1.masscenter) \
+   ...        - l1 / 2 * link1.x + l4 / 2 * link4.x
+   >>> fh = Matrix([loop.dot(link1.x), loop.dot(link1.y)])
+
+In order to generate the equations of motions, we will use the ``JointsMethod`` as our fronted. Before setting up the ``KanesMethod`` as its backend we need to calculate the velocity constraints. ::
+
+   >>> method = JointsMethod(link1, joint1, joint2, joint3)
+   >>> t = dynamicsymbols._t
+   >>> qdots = solve(method.kdes, [q1.diff(t), q2.diff(t), q3.diff(t)])
+   >>> fhd = fh.diff(t).subs(qdots)
+
+Now we can setup the ``KanesMethod`` as the backend and compute the equations of motion. ::
+
+   >>> bodylist = method._convert_bodies()
+   >>> method._method = KanesMethod(method.frame, q_ind=[q1], u_ind=[u1],
+   ...                              q_dependent=[q2, q3], u_dependent=[u2, u3],
+   ...                              kd_eqs=method.kdes, configuration_constraints=fh,
+   ...                              velocity_constraints=fhd, forcelist=method.loads,
+   ...                              bodies=bodylist)
+   >>> simplify(method.method._form_eoms())
```

Comment:
These `_` methods are effectively private and showing how they are used in the docs indicates that we might be supporting them as public API. Can you just setup a KanesMethod() separately and extract the info you need from JointsMethod and pass into the KanesMethod without attaching it to JointsMethod so that private methods are not used?

Reply:
Yeah, sure. Personally I liked to keep it under the `JointsMethod`, so you just keep one method variable in the open and than you'll also be able to use `jointsmethod.mass_matrix` etc.<br>
Also we are also using `_convert_bodies`. Not sure exactly why this method is in the `JointMethod`. You should be able to pass bodies from the `Body` class to the backend methods (opinion, not sure whether this is possible).

Reply:
> Yeah, sure. Personally I liked to keep it under the JointsMethod, so you just keep one method variable in the open and than you'll also be able to use jointsmethod.mass_matrix etc.

Sure, but we shouldn't have private methods in the documentation. The two options are: 1) don't use private methods or 2) convert methods to public. If 2) is used we should be fairly certain that is the API we want because once public, we are stuck with it.

>  Also we are also using _convert_bodies. Not sure exactly why this method is in the JointMethod. You should be able to pass bodies from the Body class to the backend methods (opinion, not sure whether this is possible).

RigidBody and Particle work with KanesMethod but I don't think Body does. So maybe that's why.

Reply:
I would personally go with a third option, namely check why `Lagrange` and/or `Kanes` cannot handle the `Body` class and fix that. Because it seems to me that `Body` should also be compatible.

Reply:
This is a related issue: https://github.com/sympy/sympy/issues/21964

Yes, you can make that fix, but if the goal is to get this example in now, then it's simpler to make the example work with the current public API and then when you improve the functionality later, this example is updated to match.